tests/test_views.py

```diff
@@ -2665,16 +2666,42 @@ def test_claim_user_when_user_is_registered_with_orcid(self, mock_response_from_
         orcid_user, validated_credentials, cas_resp = generate_external_user_with_resp(url)
         mock_response_from_ticket.return_value = authenticate(
             orcid_user,
-            cas_resp.attributes.get('accessToken', ''),
             redirect(url)
         )
         orcid_user.set_unusable_password()
         orcid_user.save()
 
-        ticket = fake.md5()
-        url += '?ticket={}'.format(ticket)
-        res = self.app.get(url)
-        res = res.follow()
+        # The request to OSF with CAS service ticket must not have cookie and/or auth.
+        service_ticket = fake.md5()
+        url_with_service_ticket = f'{url}?ticket={service_ticket}'
+        res = self.app.get(url_with_service_ticket)
+        # The response of this request is expected to be a 302 with `Location`.
+        # And the redirect URL must equal to the originial service URL
+        assert res.status_code == 302
+        redirect_url = res.headers['Location']
+        assert redirect_url == service_url
+        # The response of this request is expected have the `Set-Cookie` header with OSF cookie.
+        # And the cookie must belong to the ORCiD user.
+        raw_set_cookie = res.headers['Set-Cookie']
+        assert raw_set_cookie
+        simple_cookie = SimpleCookie()
+        simple_cookie.load(raw_set_cookie)
+        cookie_dict = {key: value.value for key, value in simple_cookie.items()}
+        osf_cookie = cookie_dict.get(settings.COOKIE_NAME, None)
```

Comment:
Note: we may want to put this into a helper method if we use if often in the future in tests.